### PR TITLE
Priority feature in Leads

### PR DESCRIPTION
**Pull request by subinasr**
Depends on https://github.com/the-deep/server/pull/477

- Segment Input in Lead Add Page
- Display Priority column in Lead Table
- Filter leads in Lead Table

### CODE REVIEW

File: src/views/Leads/index.js [hunk 1]
```diff
@@ -183,6 +183,10 @@ const tableHeadersMap = {
         label: _ts('leads', 'tableHeaderNoOfEntries'),
         sortable: true,
     },
+    priority: {
+        label: _ts('leads', 'priorityLevel'),
+        sortable: true,
```
Comment (tnagorra) [inline src/views/Leads/index.js:188]: Has this been tested?

File: src/views/LeadAdd/LeadDetail/faramSchema.js [hunk 2]
```diff
@@ -16,6 +16,7 @@ const commonFields = {
     publishedOn: [requiredCondition, dateCondition],
     sourceType: [requiredCondition],
     project: [requiredCondition],
+    priority: [],
```
Comment (tnagorra) [inline src/views/LeadAdd/LeadDetail/faramSchema.js]: Is priority not required?

File: src/views/Leads/index.js (hunk 1 above)
Comment (AdityaKhatri) [reply src/views/Leads/index.js:188]: yes

**Comment by tnagorra** [comment]
@subinasr @AdityaKhatri  Have we added priority in Grid view?

**Comment by tnagorra** [comment]
I have leads that have priority "low" by on leads page.
When I click on edit, the AddLead page entry for that lead doesn't have priority filled as "low"

Update: Looks like the problem is on the replace logic if the lead was already being edited in AddLead page.
Created an issue for this https://github.com/the-deep/deeper/issues/916

**Comment by tnagorra** [comment]
Cannot edit and save lead now.
image

This happens for entires with EMM (may not be a problem related to this PR)
Created an issue for this https://github.com/the-deep/deeper/issues/915

**Review by tnagorra — APPROVED** [review]

**Pull request merged**